src/cfnlint/rules/resources/properties/ValuePrimitiveType.py

```diff
@@ -55,30 +55,35 @@ def check_primitive_type(self, value, item_type, path):
         if item_type in ['String']:
             if not isinstance(value, (str, six.text_type, six.string_types)):
                 message = 'Property %s should be of type String' % ('/'.join(map(str, path)))
-                matches.append(RuleMatch(path, message))
+                extra_args = {'actual_type': type(value).__name__, 'expected_type': str.__name__}
+                matches.append(RuleMatch(path, message, **extra_args))
         elif item_type in ['Boolean']:
             if not isinstance(value, (bool)):
                 message = 'Property %s should be of type Boolean' % ('/'.join(map(str, path)))
-                matches.append(RuleMatch(path, message))
+                extra_args = {'actual_type': type(value).__name__, 'expected_type': bool.__name__}
+                matches.append(RuleMatch(path, message, **extra_args))
         elif item_type in ['Double']:
             if not isinstance(value, (float, int)):
                 message = 'Property %s should be of type Double' % ('/'.join(map(str, path)))
-                matches.append(RuleMatch(path, message))
+                extra_args = {'actual_type': type(value).__name__, 'expected_type': [float.__name__, int.__name__]}
+                matches.append(RuleMatch(path, message, **extra_args))
         elif item_type in ['Integer']:
             if not isinstance(value, (int)):
                 message = 'Property %s should be of type Integer' % ('/'.join(map(str, path)))
-                matches.append(RuleMatch(path, message))
+                extra_args = {'actual_type': type(value).__name__, 'expected_type': int.__name__}
+                matches.append(RuleMatch(path, message, **extra_args))
         elif item_type in ['Long']:
             if sys.version_info < (3,):
                 integer_types = (int, long,)  # pylint: disable=undefined-variable
             else:
                 integer_types = (int,)
             if not isinstance(value, integer_types):
                 message = 'Property %s should be of type Long' % ('/'.join(map(str, path)))
-                matches.append(RuleMatch(path, message))
+                matches.append(RuleMatch(path, message, **extra_args))
```

Comment:
You're missing setting extra_args in this one.  That should get your tests to green.